packages/runtime-dom/src/components/TransitionGroup.ts

```diff
@@ -138,7 +138,13 @@ const TransitionGroupImpl: ComponentOptions = {
         }
       }
 
-      return createVNode(tag, null, children)
+      return createVNode(
+        tag,
+        {
+          slotScopeIds: instance.vnode.slotScopeIds,
+        },
+        children,
+      )
```

Comment:
I don't think the props should be used to pass around the `slotScopeIds`, that's not their job. Putting it in the props can cause it to leak into the DOM. e.g.:

- [Playground](https://deploy-preview-7541--vue-sfc-playground.netlify.app/#eNp9UkFOwzAQ/IrlSwFVjRCcQqgEqELlAAh69MUk29StY1u2E1pV/Ttruy2hAk7Z3ZnZzNje0jtjRl0LNKeFK60wnjjwrRkzJRqjrScPujFkbnVDBqMsNIE+YKrIEh+Z2HhojOQesCOkCLRYYT2zXDnhhVaPVreGeF7fMlqJjtE9BUnYkhVsELjsjQmZkgXvgHBysuUozFB5+FN2Qkpeomcsi6znkQ7pIUpI3kOYD2aiFEsntR9j0vBJ2xL4Yxm2zm8kEFdqAxVO8iDwUJ1dnJNt0H3wclWjJ1XlxEJ1w9QunmCQJTsexWou6tHSaYWeoozREl0KCfbFhFiO0TwtDBiXUn8+xZm3LQwP83IB5eqX+dKtw4zRVwsObAeMHjHPbQ0+wZP3Z1hjfQQbXbUS2f+Ab+C0bIPHRLvHqGi7x4tup/FNCVXP3GTtAW9rHyoYDcxd5DOKFxMu6K/o33avRtdRhydKd1+Bfu0r)

![image](https://github.com/vuejs/core/assets/65301168/86d783eb-ee0e-46be-9d1c-f78c30c29c91)

Note the `slotscopeids` attribute in the DOM.

Perhaps it could be set after creating the node instead? e.g.:

```js
const node = createVNode(tag, null, children)
node.slotScopeIds = instance.vnode.slotScopeIds
return node
```

Reply:
Agree with @skirtles-code